OpenRA.Mods.Common/Widgets/ObserverProductionIconsWidget.cs

```diff
@@ -132,5 +169,52 @@ public override Widget Clone()
 		{
 			return new ObserverProductionIconsWidget(this);
 		}
+
+		public override void MouseEntered()
+		{
+			if (TooltipContainer != null)
+				tooltipContainer.Value.SetTooltip(TooltipTemplate,
+					new WidgetArgs() { { "player", GetPlayer() }, { "getTooltipIcon", GetTooltipIcon } });
+		}
+
+		public override void MouseExited()
+		{
+			if (TooltipContainer == null)
+				return;
+
+			tooltipContainer.Value.RemoveTooltip();
+		}
+
+		public override void Tick()
+		{
+			if (TooltipIcon != null && iconRects[lastIconIdx].Contains(Viewport.LastMousePos))
+				return;
```

Comment:
The TooltipIcon on position x is still the same - only Actor changes in Draw - null, barracks, power plant. ProductionTooltipLogic shows barracks, then shortly nothing, then power plant. At least my test confirms it.